test_driver/transitions_perf_test.dart

```diff
@@ -0,0 +1,231 @@
+}
+
+/// Returns a [Future] that resolves to true if the widget specified by [finder]
+/// is present, false otherwise.
+Future<bool> isPresent(SerializableFinder finder, FlutterDriver driver,
+    {Duration timeout = const Duration(seconds: 5)}) async {
+  try {
+    await driver.waitFor(finder, timeout: timeout);
+    return true;
+  } catch (exception) {
+    return false;
+  }
+}
+
+/// Scrolls each each demo into view, launches it, then returns to the
+/// home screen, twice.
+Future<void> runDemos(List<String> demos, FlutterDriver driver) async {
+  String currentDemoCategory;
+  SerializableFinder demoList;
+  SerializableFinder demoItem;
+
+  for (final demo in demos) {
+    if (_skippedDemos.contains(demo)) continue;
+
+    print('> $demo');
+
+    final demoCategory = demo.substring(demo.indexOf('@') + 1);
+    if (demoCategory != currentDemoCategory) {
+      // We've switched categories.
+      currentDemoCategory = demoCategory;
+      demoList = find.byValueKey('${demoCategory}DemoList');
+
+      // We may want to return to the previous category later.
+      // Reset its scroll (matters for desktop layout).
+      if (demoItem != null) await scrollToTop(demoItem, driver);
+
+      // Scroll to the category list.
+      if (demoCategory != 'study') {
+        print('scrolling to $currentDemoCategory category');
+        await driver.scrollUntilVisible(
+          homeList,
+          demoList,
+          dyScroll: -1000,
+          timeout: const Duration(seconds: 10),
+        );
+      }
+    }
+
+    // Scroll to demo and open it twice.
+    demoItem = find.byValueKey(demo);
+
+    print('scrolling to demo');
+    await driver.scrollUntilVisible(
+      demoList,
+      demoItem,
+      dxScroll: -500,
+      dyScroll: -50,
+      alignment: 0.5,
+      timeout: const Duration(seconds: 10),
+    );
+
+    for (var i = 0; i < 2; i += 1) {
+      print('tapping demo');
+      await driver.tap(demoItem); // Launch the demo
+
+      sleep(const Duration(milliseconds: 500));
+
+      if (_unsynchronizedDemos.contains(demo)) {
+        await driver.runUnsynchronized<void>(() async {
+          await driver.tap(backButton);
+        });
+      } else {
+        await driver.tap(backButton);
+      }
+    }
+    print('< Success');
+  }
+
+  await scrollToTop(demoItem, driver);
+}
+
+void main([List<String> args = const <String>[]]) {
+  group('flutter gallery transitions', () {
+    FlutterDriver driver;
+    IsolatesWorkaround workaround;
+
+    setUpAll(() async {
+      driver = await FlutterDriver.connect();
+
+      // See _handleMessages() in transitions_perf.dart.
+      _allDemos = List<String>.from(json.decode(
+        await driver.requestData('demoDescriptions'),
+      ) as List<dynamic>);
+      if (_allDemos.isEmpty) throw 'no demo names found';
+
+      // TODO: Remove workaround for https://github.com/flutter/flutter/issues/24703
+      final isWeb = await driver.requestData('isWeb') == 'true';
```

Comment:
Ok I think this makes sense now. Maybe _handleMessasages could return an enum?

Reply:
Unfortunately, can only use String https://api.flutter.dev/flutter/flutter_driver_extension/DataHandler.html

Reply:
Ah, this is good then